File: Go/Easy/Find Three Largest Numbers/solution_1.go
package main

import "math"

func FindThreeLargestNumbers(array []int) []int {

	lst := []int{math.MinInt32, math.MinInt32, math.MinInt32}

	for _, element := range array {
		updateLargest(lst, element)
	}
	return lst
}

func updateLargest(array []int, element int) {

	if element > array[2] {
		shiftAndUpdate(array, element, 2)
	} else if element > array[1] {
		shiftAndUpdate(array, element, 1)
	} else if element > array[0] {
		shiftAndUpdate(array, element, 0)
	}

}

func shiftAndUpdate(array []int, element int, idx int) {

	for i := 0; i < idx+1; i++ {

		if i == idx {
			array[i] = element
		} else {
			array[i] = array[i+1]
		}
	}
}

File: Python/Easy/Find Three Largest Numbers/solution_1.py
#Hard Coded
def findThreeLargestNumbers(array) -> list:

    lst = [None,None,None]

    for element in array:

        if lst[2] == None or element > lst[2]:
            lst[0] = lst[1]
            lst[1] = lst[2]
            lst[2] = element
        elif lst[1] == None or element > lst[1]:
            lst[0] = lst[1]
            lst[1] = element
        elif lst[0] == None or element > lst[0]:
            lst[0] = element
        
    return lst
